File: esports-backend/src/main/java/com/esports/dto/LoginRequest.java
package com.esports.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class LoginRequest {
    @NotBlank(message = "Email is required")
    @Pattern(
        regexp = "^[a-zA-Z0-9._%+-]+@(gmail\\.com|admin\\.com)$",
        message = "Email must be from gmail.com domain"
    )
    private String email;

    @NotBlank(message = "Password is required")
    private String password;
}

File: esports-backend-dotnet/Upkart1/DTO/LoginRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Upkart1.DTO
{
    public class LoginRequest
    {
        [Required(ErrorMessage = "Email is required")]
        [RegularExpression(@"^[a-zA-Z0-9._%+-]+@(gmail\.com|admin\.com)$", ErrorMessage = "Email must be from gmail.com domain")]
        [JsonPropertyName("email")]
        public string Email { get; set; }

        [Required(ErrorMessage = "Password is required")]
        [JsonPropertyName("password")]
        public string Password { get; set; }
    }
}
